build.gradle.kts

```diff
@@ -30,24 +24,24 @@ plugins {
 	id("org.ajoberstar.grgit")
 }
 
-operator fun String.invoke(): String {
-	return (rootProject.properties[this] as String?)!!
-}
+operator fun String.invoke(): String = rootProject.properties[this] as? String ?: error("Property $this not found")
 
 enum class ReleaseChannel(
 	val suffix: String? = null,
 	val releaseType: ReleaseType? = null,
-	val compress: Boolean,
+	val compress: CompressionType
 	) {
-	DEV_BUILD(suffix = "dev", compress = false),
-	PRE_RELEASE(suffix = "pre", compress = false),
-	RELEASE_CANDIDATE(suffix = "rc", compress = true),
-	RELEASE(releaseType = ReleaseType.STABLE, compress = true),
+	DEV_BUILD(suffix = "dev", compress = CompressionType.NONE),
+	PRE_RELEASE(suffix = "pre", compress = CompressionType.NONE),
+	RELEASE_CANDIDATE(suffix = "rc", compress = CompressionType.LIBDEFLATE),
+	RELEASE(releaseType = ReleaseType.STABLE, compress = CompressionType.SEVENZIP),
```

Comment:
Compression settings for release candidate and release should match. The reason I don't compress lower versions is so it doesnt strip the LVTs and source files in builds I anticipate wanting to read stack traces for.

Reply:
i'll split up the lossy compression settings and the deflation into two separate things for more fine-grained control.

Reply:
Sounds good. Feel free to move the settings out of `gradle.properties` if that makes the implementation saner. You can also add lossless compression for builds earlier in the release process if you want.